scrapper.py

```diff
@@ -21,77 +39,245 @@ class IncorrectNumberOfArticlesError(Exception):
+                    class_urls = page_soup.find('ul',
+                                                class_='panel-menu-list list-unstyled text-uppercase').find_all('a')
+                    for tag_a_content in class_urls:
+                        found_link = tag_a_content.get('href')
+                        if found_link not in self.seed_urls:
+                            self.seed_urls.append(found_link)
+                else:
+                    found_links = self._extract_url(page_soup)
+                    new_links = [link for link in found_links if link not in self.seed_urls]
+                    self.seed_urls.extend(new_links)
+        with open(os.path.join(PROJECT_ROOT, 'saved_urls.json'), encoding='utf-8') as file:
+            saved_urls = json.load(file)
+        saved_urls['seed_urls'] = self.seed_urls
+        with open(os.path.join(PROJECT_ROOT, 'saved_urls.json'), 'w', encoding='utf-8') as file:
+            json.dump(saved_urls, file)
+
+        return self.seed_urls
 
 
 class ArticleParser:
     """
     ArticleParser implementation
     """
+
     def __init__(self, full_url: str, article_id: int):
-        pass
+        self.full_url = full_url
+        self.article_id = article_id
+        self.article = article.Article(full_url, article_id)
 
     def _fill_article_with_text(self, article_soup):
-        pass
+        paragraphs_soup = article_soup.find('div', class_='element-detail')
+        article_list = [paragraph.strip() for paragraph in paragraphs_soup.text.split('\n') if paragraph.strip()]
+        article_str = ' '.join(article_list)
+        self.article.text = article_str
 
     def _fill_article_with_meta_information(self, article_soup):
-        pass
+        self.article.title = article_soup.find('h1').text.strip()
+        existed_author = article_soup.find('div', class_='text-box text-right').find('a').text.strip()
+        if existed_author:
+            self.article.author = existed_author
+        else:
+            self.article.author = 'NOT FOUND'
+        self.article.date = self.unify_date_format(article_soup.find('p', class_='date font-open-s-light').text)
 
     @staticmethod
     def unify_date_format(date_str):
         """
         Unifies date format
         """
-        pass
+        date, time = date_str.split()
+        day, month, year = date.split('.')
+        hours, minutes, secs = time.split(':')
+        date_int = tuple(map(int, [year, month, day, hours, minutes, secs]))
+        valid_date = datetime(*date_int)
+        return valid_date
 
     def parse(self):
         """
         Parses each article
         """
-        pass
+        response = requests.get(self.full_url, headers=headers)
+        if response:
+            article_soup = BeautifulSoup(response.content, features='lxml')
+            self._fill_article_with_text(article_soup)
+            self._fill_article_with_meta_information(article_soup)
+        self.article.save_raw()
```

Comment:
remember S in SOLID? is it responsibility of **parse** method to **save** the file?